crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -1433,12 +1433,46 @@ impl<F: Clone + Sync> UpdateTracker<F, PaymentData<F>, api::PaymentsRequest> for
             None => (None, None, None),
         };
 
-        let payment_attempt_fut = tokio::spawn(
+        let pmt_order_tax_amount =
+            payment_data
+                .payment_intent
+                .tax_details
+                .as_ref()
+                .and_then(|tax_details| {
+                    tax_details
+                        .payment_method_type
+                        .as_ref()
+                        .map(|payment_method_tax| payment_method_tax.order_tax_amount)
+                        .or_else(|| {
+                            tax_details
+                                .default
+                                .as_ref()
+                                .map(|default_tax| default_tax.order_tax_amount)
+                        })
+                });
+
+        let net_amount = hyperswitch_domain_models::payments::payment_attempt::NetAmount::new(
+            payment_data.payment_attempt.net_amount.get_order_amount(),
+            payment_data.payment_intent.shipping_cost,
+            pmt_order_tax_amount,
+            surcharge_amount,
+            tax_amount,
+        );
+
+        let payment_attempt_fut: tokio::task::JoinHandle<
+            Result<
+                hyperswitch_domain_models::payments::payment_attempt::PaymentAttempt,
+                error_stack::Report<
+                    hyperswitch_domain_models::errors::api_error_response::ApiErrorResponse,
+                >,
+            >,
+        > = tokio::spawn(
             async move {
                 m_db.update_payment_attempt_with_attempt_id(
                     m_payment_data_payment_attempt,
                     storage::PaymentAttemptUpdate::ConfirmUpdate {
                         currency: payment_data.currency,
+                        order_tax_amount: net_amount.get_order_tax_amount(),
```

Comment:
can you remove this? we are already updating this via the net amount field right